### ISSUE TITLE
Synology Install Error

### ISSUE THREAD

**Issue by ammacdonald3** [daledavies/jump#51]
Hi, I've been able to install Jump without issue on my MacBook, but when I install to my Synology NAS using the same .yml file, I receive the error below.  The error only appears after trying to navigate to the startpage in the browser on port 8123.  Additionally, when installing to my Synology, Docker kept erroring out saying that each folder (backgrounds, favicon, search, sites) did not exist.  I manually created them, and while it seemed to install correctly after that, the below error was logged when trying to actually use the tool.  Thanks in advance for any insight!

NOTICE: PHP message: PHP Warning:  file_get_contents(/var/www/html/sites/sites.json): Failed to open stream: No such file or directory in /var/www/html/classes/Sites.php on line 75
172.22.0.1 - - [29/Jul/2022:18:28:47 +0000] "GET / HTTP/1.1" 500 5 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36" "-" 0.164 0.164 . -
NOTICE: PHP message: PHP Fatal error:  Uncaught Jump\Exceptions\ConfigException: There was a problem loading the sites.json file in /var/www/html/classes/Sites.php:77
Stack trace:
#0 /var/www/html/classes/Sites.php(48): Jump\Sites->load_sites_from_json()
#1 /var/www/html/classes/Cache.php(141): Jump\Sites->Jump\{closure}()
#2 /var/www/html/vendor/nette/caching/src/Caching/Cache.php(105): Jump\Cache->Jump\{closure}()
#3 /var/www/html/classes/Cache.php(142): Nette\Caching\Cache->load()
#4 /var/www/html/classes/Sites.php(49): Jump\Cache->load()
#5 /var/www/html/classes/Pages/HomePage.php(42): Jump\Sites->__construct()
#6 /var/www/html/classes/Pages/AbstractPage.php(60): Jump\Pages\HomePage->render_header()
#7 /var/www/html/classes/Pages/AbstractPage.php(67): Jump\Pages\AbstractPage->build_page()
#8 /var/www/html/classes/Main.php(69): Jump\Pages\AbstractPage->get_output()
#9 /var/www/html/index.php(22): Jump\Main->init()
#10 {main}
  thrown in /var/www/html/classes/Sites.php on line 77

**Comment by daledavies** [comment]
See if you can check in the container if you really do have a sites.json file mounted there, if so what are the permissions etc.

I don't know what difference the Synology NAS makes but docker-compose should create the mapped directories locally if they don't exist, so something has gone wrong there if you had to create them yourself.

**Comment by daledavies** [comment]
It really sounds like the directories have the wrong permissions, so the container can't create the files in them when the entrypoint script runs.

**Comment by ammacdonald3** [comment]
Thanks so much for the quick reply!  Based on your feedback, I was able to get it running by manually copying over the sites.json file from my clean MacBook install as it was missing from my Synology install.  The sites' icon files were also missing, so I copied those over too.  Interestingly, the background images and the favicon file all installed correctly directly on my Synology.  I also originally executed the docker compose command with sudo, so I'm not sure why I wouldn't have had permissions to create these certain objects.  Closing out the issue now as it's currently working for me, but let me know if I can provide any more info about my environment in case you see this again.
